projectHandler.js

```diff
@@ -68,84 +68,100 @@ export async function retrieve(event, context, callback) {
-		ExpressionAttributeNames: {
-		'#projectOwner': 'projectOwner',
-		},
-		ExpressionAttributeValues: {
-        ':userId': inputParams.userId,		
-		},
+		TableName: process.env.userstableName,
+		Key: {
+			userId: event.queryStringParameters.userId
+		}
 	};
-	try {		
-		dynamoDb.scan(params, function(err,data){
-			if(err){
-				//console.log(userId);
-				console.log(err);
-				callback(err,null);
-			}else{
-				console.log(data);
-				callback(null, success(data));
+	try {
+		console.log(params);
+		const result = await dynamoDbLib.call("get", params);
+		console.log(result);
+		if (result.Item) {
+			// Return the retrieved item
+			const dynamoDb = new AWS.DynamoDB.DocumentClient();
+			var projectparams;
+			if(result.Item.userRole == "manager") {
+			projectparams = {
+			TableName: process.env.projectstableName,
+			FilterExpression: ' projectOwner = :userId',
+			ExpressionAttributeValues: {
+			':userId': event.queryStringParameters.userId		
 			}
-		});
-	} catch (e) {
-		console.log(e);
-		callback(null, failure({ status: false }));
-	}
-}
-
-//Lists all projects of the developer working on
-export async function listDeveloperProjects(event, context, callback) {
-	const dynamoDb = new AWS.DynamoDB.DocumentClient();
-	const inputParams = JSON.parse(event.queryStringParameters);	
-	console.log(inputParams.userId);	
-	const params = {
-		TableName: process.env.projectstableName,
-		FilterExpression: 'contains (#projectContributors, :userId)',
-		ExpressionAttributeNames: {
-		'#projectContributors': 'projectContributors',
-		},
-		ExpressionAttributeValues: {
-        ':userId': inputParams.userId	
-		},
-	};
-		
-	try {		
-		dynamoDb.scan(params, function(err,data){
-			if(err){
-				console.log(err);
-				callback(err,null);
+		};
+		}else if(result.Item.userRole=="developer") {
+			projectparams = {
+			TableName: process.env.projectstableName,
+			FilterExpression: 'contains (projectContributors, :userId)',
+			ExpressionAttributeValues: {
+			':userId': event.queryStringParameters.userId		
+			}
+		};
+		}
+		try {
+			console.log(projectparams);
+			const scanResult = await dynamoDb.scan(projectparams).promise();
+			if(scanResult){
+			console.log(scanResult);
+			callback(null, success(scanResult));
+				//console.log(userId);				
 			}else{
-				
-				console.log(data);
-				callback(null, success(data));
+			console.log(err);
+			callback(err,null);	
 			}
-		});
-	} catch (e) {
+		} catch (e) {
 		console.log(e);
 		callback(null, failure({ status: false }));
+	}}
+	
+	}catch (e) {
+	callback(null, failure({ status: false , error: "User does not exist." })); 
+	}
 }
-}
-
 
 //Deletes project based on the projectid specified
 export async function deleteProject(event, context, callback) {
 	const params = {
 		TableName: process.env.projectstableName,
 		Key: {
 			projectId: event.pathParameters.id
-		}
+		},
+		ReturnValues: 'ALL_OLD'
 	};
 
 	try {
 		const result = await dynamoDbLib.call("delete", params);
```

Comment:
We should check here if user that calls the api has permission to delete object before deleting it

Reply:
Return 401 Unauthorized if user does not have proper permissions

Reply:
Yup this code will be changed further, as i still have to delete project entry from user table also. Nav and myself, we thought we will have a discussion once he starts implementing delete. As deletion gets trickier based on some of the scenarios like:
1.If there are some of the pending tasks then that also should be forcefully deleted.
2.All these deletes(task/project) has to be updated in users table also.

Reply:
Cool, as long as we're keeping track of this